inject/__init__.py

```diff
@@ -266,6 +266,29 @@ def __call__(self) -> T:
         return self._instance
 
 
+class _NullInjectedAttribute(object):
+    """
+    Returned for injected attributes that don't have an injection configuration.
+    """
+    pass
+
+
+class _AttributeInjection(object):
+    def __init__(self, cls):
+        self._cls = cls
+
+    def __get__(self, obj, owner):
+        # Some Python internals (starting in 3.8 specifically) may call
+        # getattr() for a class's attributes to do some reflective inspection.
+        # That triggers this codepath, potentially before inject.configure()
+        # has been called, which is why we should silently return a sentinel
+        # type representing a null value instead of raising.
+        try:
+            return instance(self._cls)
+        except InjectorException:
+            return _NullInjectedAttribute()
+
+
 class _AttributeInjection(object):
```

Comment:
@andrewborba10 I was curious to see your changes, and it made me wonder one thing.

You added a new `_AttributeInjection` class like below:
```
class _NullInjectedAttribute(object):
    """
    Returned for injected attributes that don't have an injection configuration.
    """
    pass


class _AttributeInjection(object):
    def __init__(self, cls):
        self._cls = cls

    def __get__(self, obj, owner):
        # Some Python internals (starting in 3.8 specifically) may call
        # getattr() for a class's attributes to do some reflective inspection.
        # That triggers this codepath, potentially before inject.configure()
        # has been called, which is why we should silently return a sentinel
        # type representing a null value instead of raising.
        try:
            return instance(self._cls)
        except InjectorException:
            return _NullInjectedAttribute()
```

It seems to me that now we should delete this `_AttributeInjection` class that does not return the sentinel value.

Am I missing something?

Reply:
That was definitely unintentional, I meant to modify the existing class. It sounds like Ivan might have some other changes in mind, so I'll let him apply those with the fix to remove the old class.